src/app/Dashboard/AutomatedAnalysis/AutomatedAnalysisCard.tsx

```diff
@@ -0,0 +1,724 @@
+      setErrorMessage(errorMessage);
+      setIsError(true);
+      setIsLoading(false);
+      setUsingArchivedReport(false);
+      setUsingCachedReport(false);
+    },
+    [setErrorMessage, setIsError, setIsLoading, setUsingArchivedReport, setUsingCachedReport]
+  );
+
+  const handleLoading = React.useCallback(() => {
+    setIsLoading(true);
+    setIsError(false);
+    setUsingArchivedReport(false);
+    setUsingCachedReport(false);
+  }, [setIsLoading, setIsError, setUsingArchivedReport, setUsingCachedReport]);
+
+  const handleArchivedRecordings = React.useCallback(
+    (recordings: ArchivedRecording[]) => {
+      const freshestRecording = recordings.reduce((prev, current) =>
+        prev?.archivedTime > current?.archivedTime ? prev : current
+      );
+      addSubscription(
+        context.target.target().subscribe((target) => {
+          context.reports
+            .reportJson(freshestRecording, target.connectUrl)
+            .pipe(first())
+            .subscribe({
+              next: (report) => {
+                setUsingArchivedReport(true);
+                setReportTime(freshestRecording.archivedTime);
+                categorizeEvaluation(report);
+              },
+              error: () => {
+                handleStateErrors(INTERNAL_ERROR_MESSAGE);
+              },
+            });
+        })
+      );
+    },
+    [
+      addSubscription,
+      context.target,
+      context.reports,
+      categorizeEvaluation,
+      handleStateErrors,
+      setUsingArchivedReport,
+      setReportTime,
+    ]
+  );
+
+  const handleEmptyRecordings = React.useCallback(
+    (connectUrl: string) => {
+      const cachedReportAnalysis = context.reports.getCachedAnalysisReport(connectUrl);
+      if (cachedReportAnalysis.report.length > 0) {
+        setUsingCachedReport(true);
+        setReportTime(cachedReportAnalysis.timestamp);
+        categorizeEvaluation(cachedReportAnalysis.report);
+      } else {
+        addSubscription(
+          queryArchivedRecordings(connectUrl)
+            .pipe(
+              first(),
+              map((v) => v.data.archivedRecordings.data as ArchivedRecording[])
+            )
+            .subscribe({
+              next: (recordings) => {
+                if (recordings.length > 0) {
+                  handleArchivedRecordings(recordings);
+                } else {
+                  handleStateErrors(NO_RECORDINGS_MESSAGE);
+                }
+              },
+              error: () => {
+                handleStateErrors(NO_RECORDINGS_MESSAGE);
+              },
+            })
+        );
+      }
+    },
+    [
+      addSubscription,
+      context.reports,
+      categorizeEvaluation,
+      queryArchivedRecordings,
+      handleArchivedRecordings,
+      handleStateErrors,
+      setUsingCachedReport,
+      setReportTime,
+    ]
+  );
+
+  const generateReport = React.useCallback(() => {
+    addSubscription(
+      context.target.target().subscribe((target) => {
```

Comment:
This implicitly also subscribes to target change. We should move this out of this function into a useEffect. Maybe replace:

```typescript
React.useEffect(() => {
  generateReport();
}, []);
```

with:

```typescript
React.useEffect(() => {
  context.target.target().subscribe((target) => {
    setTargetConnectURL(target.connectUrl);
    dispatch(automatedAnalysisAddTargetIntent(target.connectUrl));
    generateReport();
  });
}, [context.target, generateReport, setTargetConnectURL, dispatch]);
```

And within generateReport, we can call context.target.target again, but with `filter` valid target and use `first`.